irohad/torii/impl/command_service_impl.cpp

```diff
@@ -150,6 +152,27 @@ namespace iroha {
 
     void CommandServiceImpl::processBatch(
         std::shared_ptr<shared_model::interface::TransactionBatch> batch) {
+      auto cache_presence = tx_presence_cache_->check(*batch);
```

Comment:
I suggest to make a look up in `cache_` for response first, then access `tx_presence_cache_`. If there was no entry in `cache_` for the transaction, but `tx_presence_cache_` returns `Committed` or `Rejected`, it can be pushed into status bus. In this case there will be no problem testing the replay in ITF.